src/retractions.jl

```diff
@@ -0,0 +1,253 @@
+            m,
+            X0,
+            project_point,
+            project_tangent,
+            nlsolve_kwargs,
+        )
+    end
+end
+function NLsolveInverseRetraction(
+    m,
+    X0 = nothing;
+    project_tangent::Bool = false,
+    project_point::Bool = false,
+    nlsolve_kwargs...,
+)
+    return NLsolveInverseRetraction(m, X0, project_point, project_tangent, nlsolve_kwargs)
+end
+
+
+"""
+    inverse_retract!(M::Manifold, X, p, q[, method::AbstractInverseRetractionMethod])
+
+Compute the inverse retraction, a cheaper, approximate version of the
+[`log`](@ref)arithmic map), of points `p` and `q` on the [`Manifold`](@ref) `M`.
+Result is saved to `X`.
+
+Inverse retraction method can be specified by the last argument, defaulting to
+[`LogarithmicInverseRetraction`](@ref). See the documentation of respective manifolds for
+available methods.
+
+See also [`retract!`](@ref).
+"""
+function inverse_retract!(M::Manifold, X, p, q)
+    return inverse_retract!(M, X, p, q, LogarithmicInverseRetraction())
+end
+function inverse_retract!(M::Manifold, X, p, q, method::LogarithmicInverseRetraction)
+    return log!(M, X, p, q)
+end
+function inverse_retract!(M::Manifold, X, p, q, method::AbstractInverseRetractionMethod)
+    return error(
+        manifold_function_not_implemented_message(M, inverse_retract!, X, p, q, method),
+    )
+end
+
+"""
+    inverse_retract(M::Manifold, x, y)
+    inverse_retract(M::Manifold, x, y, method::AbstractInverseRetractionMethod
+
+Compute the inverse retraction, a cheaper, approximate version of the
+[`log`](@ref)arithmic map), of points `p` and `q` on the [`Manifold`](@ref) `M`.
+
+Inverse retraction method can be specified by the last argument, defaulting to
+[`LogarithmicInverseRetraction`](@ref), since the [`log`](@ref)arithmic map is the inverse of a
+retraction, namely the [`exp`](@ref)onential map.
+For available invere retractions on certain manifolds see the documentation on the
+correspnonding manifold.
+
+See also [`retract`](@ref).
+"""
+function inverse_retract(M::Manifold, p, q)
+    X = allocate_result(M, inverse_retract, p, q)
+    inverse_retract!(M, X, p, q)
+    return X
+end
+function inverse_retract(M::Manifold, p, q, method::AbstractInverseRetractionMethod)
+    X = allocate_result(M, inverse_retract, p, q)
+    inverse_retract!(M, X, p, q, method)
+    return X
+end
+
+@doc raw"""
+    retract(M::Manifold, p, X)
+    retract(M::Manifold, p, X, t::Real=1)
+    retract(M::Manifold, p, X, method::AbstractRetractionMethod)
+    retract(M::Manifold, p, X, t::Real=1, method::AbstractRetractionMethod)
+
+Compute a retraction, a cheaper, approximate version of the [`exp`](@ref)onential map,
+from `p` into direction `X`, scaled by `t`, on the [`Manifold`](@ref) `M`.
+
+A retraction ``\operatorname{retr}_p: T_p\mathcal M`` is a smooth map that fulfills
+
+1. ``\operatorname{retr}_p(0) = p``
+2. ``DR_p(0): T_p\mathcal M \to T_p\mathcal M`` is the identity map, i.e. ``DR_p(0)[X]=X``.
```

Comment:
I think you mix here `\operatorname{retr}` with `R`, and you could add a note about what the `D` is.

Reply:
Thanks I added both.